Mathlib/CategoryTheory/Sites/EffectiveEpimorphic.lean

```diff
@@ -567,4 +569,49 @@ lemma effectiveEpi_iff_epi {X Y : C} (f : X ⟶ Y) : EffectiveEpi f ↔ Epi f :=
 
 end Epi
 
+section Regular
+
+open RegularEpi in
+/-- The data of an `EffectiveEpi` structure on a `RegularEpi`. -/
+noncomputable def effectiveEpiStructOfRegularEpi {B X : C} (f : X ⟶ B) [RegularEpi f] :
+    EffectiveEpiStruct f where
+  desc e h := (Cofork.IsColimit.existsUnique isColimit e (h _ _ w)).choose
+  fac e h := (Cofork.IsColimit.existsUnique isColimit e (h _ _ w)).choose_spec.1
+  uniq e h _ hg := (Cofork.IsColimit.existsUnique isColimit e (h _ _ w)).choose_spec.2 _ hg
```

Comment:
This would look slightly nicer if combine with the cofork API:
```suggestion
  desc _ h := Cofork.IsColimit.desc isColimit _ (h _ _ w)
  fac _ _ := Cofork.IsColimit.π_desc' isColimit _ _
  uniq _ _ _ hg := Cofork.IsColimit.hom_ext isColimit (hg.trans
    (Cofork.IsColimit.π_desc' _ _ _).symm)
```